src/Domain/Infra/Doctrine/MappingConfig.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\Domain\Infra\Doctrine;
+
+/**
+ * @author Pascal Wacker <hello@pascalwacker.ch>
+ *
+ * @internal
+ */
+final class MappingConfig implements MappingConfigInterface
+{
+    private $mappingFiles;
+    private $mappingDir;
+    private $keyMaxLength;
+
+    public function __construct(array $mappingFiles, array $mappingConfig = [], string $projectDir)
+    {
+        $this->mappingFiles = $mappingFiles;
+        $this->_init($mappingConfig, $projectDir);
+    }
+
+    /**
+     * @return array
+     */
+    public function getMappingFiles(): array
+    {
+        return $this->mappingFiles;
+    }
+
+    /**
+     * @return string
+     */
+    public function getMappingDir(): string
+    {
+        return $this->mappingDir;
+    }
+
+    /**
+     * @return int
+     */
+    public function getKeyMaxLength(): int
+    {
+        return $this->keyMaxLength;
+    }
+
+    /**
+     * Replaces config values in template and returns it
+     *
+     * @param string $contents
+     * @return string
+     */
+    public function interpolate (string $contents): string
+    {
+        // call handle function
+        $contents = $this->_interpolate($contents, $this);
```

Comment:
same approach, just support interpolating the know configs. There's no need to support mappingFiles and mappingDir really.. so for now could be as simply as

```php
return str_replace('%msgphp.doctrine.mapping_config.key_max_length%',  $this->keyMaxLength ?? 191);
```